docs/modules/ROOT/pages/api/account.adoc

```diff
@@ -0,0 +1,158 @@
+* xref:#Account-set_public_key[`++set_public_key(new_public_key)++`]
+* xref:#Account-get_public_key[`++get_public_key()++`]
+
+[.contract-subindex-inherited]
+.ISRC6
+
+* xref:#Account-\\__execute__[`++__execute__(calls)++`]
+* xref:#Account-\\__validate__[`++__validate__(calls)++`]
+* xref:#Account-is_valid_signature[`++is_valid_signature(hash, signature)++`]
+
+[.contract-subindex-inherited]
+.ISRC5
+
+* xref:#Account-supports_interface[`++supports_interface(interface_id)++`]
+--
+
+[.contract-index]
+.Events
+--
+* xref:#Account-OwnerAdded[`++OwnerAdded(new_owner_guid)++`]
+* xref:#Account-OwnerRemoved[`++OwnerRemoved(removed_owner_guid)++`]
+--
+
+[.contract-item]
+[[Account-__execute__]]
+==== `[.contract-item-name]#++__execute__++#++(calls: Array<Call>) → Array<Span<felt252>>++` [.item-kind]#external#
+
+See xref:ISRC6-\\__execute__[ISRC6::\\__execute__].
+
+[.contract-item]
+[[Account-__validate__]]
+==== `[.contract-item-name]#++__validate__++#++(calls: Array<Call>) → felt252++` [.item-kind]#external#
+
+See xref:ISRC6-\\__validate__[ISRC6::\\__validate__].
+
+[.contract-item]
+[[Account-__validate_declare__]]
+==== `[.contract-item-name]#++__validate_declare__++#++(class_hash: felt252) → felt252++` [.item-kind]#external#
+
+Validates a Declare transaction.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract-item]
+[[Account-__validate_deploy__]]
+==== `[.contract-item-name]#++__validate_deploy__++#++(class_hash: felt252, contract_address_salt: felt252, _public_key: felt252) → felt252++` [.item-kind]#external#
+
+Validates a DeployAccount transaction.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract-item]
+[[Account-set_public_key]]
+==== `[.contract-item-name]#++set_public_key++#++(new_public_key: felt252)++` [.item-kind]#external#
+
+Sets a new public key for the account.
+
+[.contract-item]
+[[Account-get_public_key]]
+==== `[.contract-item-name]#++get_public_key++#++()++ → felt252` [.item-kind]#external#
+
+Returns the current public key of the account.
+
+[.contract-item]
+[[Account-is_valid_signature]]
+==== `[.contract-item-name]#++is_valid_signature++#++(hash: felt252, signature: Array<felt252>) → felt252++` [.item-kind]#external#
+
+See xref:ISRC6-is_valid_signature[ISRC6::is_valid_signature].
+
+[.contract-item]
+[[Account-supports_interface]]
+==== `[.contract-item-name]#++supports_interface++#++(interface_id: felt252) → bool++` [.item-kind]#external#
+
+Returns whether a contract implements a given interface or not.
+
+[.contract-item]
+[[Account-OwnerAdded]]
+==== `[.contract-item-name]#++OwnerAdded++#++(new_owner_guid: felt252)++` [.item-kind]#event#
+
+Emitted when the account `public_key` is updated.
+
+[.contract-item]
+[[Account-OwnerRemoved]]
+==== `[.contract-item-name]#++OwnerRemoved++#++(removed_owner_guid: felt252)++` [.item-kind]#event#
+
+Emitted when the account `public_key` is updated except for initialization.
```

Comment:
>Shouldn't we include an API for internals? Or is it too big of an endeavor for this milestone you think? I only ask because that's currently included in the drafted ERC20 and 721 doc PRs

I think we should add it, yes, but maybe for the next iteration? I think having external and events is a big improvement already.

>Even if we stick with the bare bones API for now, I think we should include constructor

+1 to this.